### PR TITLE
Snyk SBOM workflow add

### PR DESCRIPTION
**Pull request by SadFaceSmith**
What this PR does / why we need it:
This PR adds a short GH Action to generate and upload SBOM reports via Snyk.

Which issue(s) this PR fixes:

Fixes #

Special notes for your reviewer:

FYSA @grafana/security-team

### CODE REVIEW

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        continue-on-error: true
+        uses: snyk/actions/golang@master
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          command: test
+          args: --json --all-projects --json-file-output=${{ github.event.repository.name }}.json  --strict-out-of-sync=false
+
+      - name: install snyk-to-html
+        run: |
```
Comment (MasslessParticle) [inline .github/workflows/snyk.yml:36]: I notice this task was run as part of the PR. This is a public CI, can a user [do this](https://www.ntweekly.com/2021/07/23/print-single-or-all-environment-variables-with-github-actions/) to dump sensitive information?

Comment (SadFaceSmith) [reply .github/workflows/snyk.yml:36]: @MasslessParticle I would not think so, the secrets in the workflow are stored in the GitHub Org. Similar to the other workflows in this repo.

**Review by MasslessParticle — APPROVED** [review]
LGTM

**Review by kavirajk — APPROVED** [review]
LGTM. thanks @SadFaceSmith.

**Pull request merged**